core/test/test_serial.cpp

```diff
@@ -176,6 +183,28 @@ TEST(ConnectConnect, PruningForward) {
 	EXPECT_EQ(c2->calls_, 6u);
 }
 
+TEST(ConnectConnect, PruningMultiForward) {
+	GeneratorMockup::id_ = Connect::id_ = 0;  // reset IDs
+	Task t;
+
+	t.setRobotModel(getModel());
+	t.add(Stage::pointer(new GeneratorMockup()));
+	t.add(Stage::pointer(new Connect()));
+	t.add(Stage::pointer(new BackwardMockup()));
+	/* TODO(v4hn):
+	 * t.add(Stage::pointer(new GeneratorMockup()));
+	 * should be enough instead of the above, but propagators do not even respect ENABLED/DISABLED yet
+	 */
+
+	t.add(Stage::pointer(new GeneratorMockup()));
+	t.add(Stage::pointer(new ForwardMockup({ 0, 0 }, 2)));  // spawn two solutions for the only incoming state
+	t.add(Stage::pointer(new ForwardMockup({ 0, inf })));
```

Comment:
This looks overly complex. Could you add a short comment describing the exact test case.

Reply:
You are one to talk, the other pruning tests feel just as complex to me. :D

I will try to address the TODO here soon and thus simplify the test. Also I just noticed the comment is wrong and should read `t.add(Stage::pointer(new BackwardMockup()));`

Reply:
> You are one to talk, the other pruning tests feel just as complex to me.

Touche.